File: C++/CLRS/Selection_Sort.cpp
#include <bits/stdc++.h>
using namespace std;

void Sel_Sort(int A[],int n)
{
    int i,j;
    for (i=0;i<n-1;i++)
    {
        int min_pos=i;
        for (j=i+1;j<n;j++)
        {
            if(A[min_pos]>=A[j])
            {
                min_pos=j;
            }
        }
        int temp=A[i];
        A[i]=A[min_pos];
        A[min_pos]=temp;
    }
}

int main()
{
    int A[] = {5, 8, 6, 4};
    int i;

    Sel_Sort(A, 4);
    for (i=0;i<4;i++)
    {
        cout << A[i] << ' ' ;
    }

    return 0;
}

File: Python/CLRS/Selection Sort.py
def Selection_Sort(A: list[int]):
    for i in range(len(A)-1):
        min_pos = i
        for j in range(i+1, len(A)):
            if A[min_pos] >= A[j]:
                min_pos = j

        temp = A[i]
        A[i] = A[min_pos]
        A[min_pos] = temp

    return A

def Sel_Sort(A: list[[int]]):
    for i in range(len(A)-1):
        m = min(A[i:len(A)])
        A.remove(m)
        A.insert(i, m)

    return A

print(Selection_Sort([2,6,8,4]))
print(Sel_Sort([3,4,1,2]))